tests/phpunit/tests/blocks/blockHooks.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Tests for block hooks feature functions.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ *
+ * @since 6.4.0
+ *
+ * @group blocks
+ */
+class Tests_Blocks_BlockHooks extends WP_UnitTestCase {
+
+	/**
+	 * Tear down after each test.
+	 *
+	 * @since 6.4.0
+	 */
+	public function tear_down() {
+		$registry = WP_Block_Type_Registry::get_instance();
+
+		foreach ( array( 'tests/my-block', 'tests/my-container-block' ) as $block_name ) {
+			if ( $registry->is_registered( $block_name ) ) {
+				$registry->unregister( $block_name );
+			}
+		}
+
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 59313
+	 *
+	 * @covers ::get_hooked_blocks
+	 */
+	public function test_get_hooked_blocks_no_match_found() {
+		$result = get_hooked_blocks( 'tests/no-hooked-blocks' );
+
+		$this->assertSame( array(), $result );
+	}
+
+	/**
+	 * @ticket 59313
+	 *
+	 * @covers ::get_hooked_blocks
+	 *
+	 * @dataProvider data_block_hooked_blocks
+	 *
+	 * @param string $block_name Block name.
+	 * @param array  $expected   Expected hooked blocks.
+	 */
+	public function test_get_hooked_blocks_matches_found( $block_name, $expected ) {
+		register_block_type(
+			'tests/my-block',
+			array(
+				'block_hooks' => array(
+					'tests/hooked-before' => 'before',
+					'tests/hooked-after'  => 'after',
+				),
+			)
+		);
+		register_block_type(
+			'tests/my-container-block',
+			array(
+				'block_hooks' => array(
+					'tests/hooked-before'      => 'before',
+					'tests/hooked-after'       => 'after',
+					'tests/hooked-first-child' => 'first_child',
+					'tests/hooked-last-child'  => 'last_child',
+				),
+			)
+		);
+
+		$this->assertSame(
+			$expected,
+			get_hooked_blocks( $block_name )
+		);
```

Comment:
TBH I'm not sure a data provider is quite the right fit here. The way I understand it, a data provider makes sense to de-duplicate both setup and assertions across a set of different values; but in our case, the setup (hooked block registration) is always the same, and all of it is needed for all assertions. So I'd prefer if we inlined all assertions here in the test 😊

Reply:
I had it like it before, I guess it's fine the way you described it with assertion messages. I will change it.